src/core/qgsellipsoidutils.cpp

```diff
@@ -186,12 +344,65 @@ QList<QgsEllipsoidUtils::EllipsoidDefinition> QgsEllipsoidUtils::definitions()
   sDefinitionCacheLock.unlock();
 
   sDefinitionCacheLock.lockForWrite();
+  QList<QgsEllipsoidUtils::EllipsoidDefinition> defs;
+
+#if PROJ_VERSION_MAJOR>=6
+  sEllipsoidCacheLock.lockForWrite();
+
+  PJ_CONTEXT *context = QgsProjContext::get();
+  PROJ_STRING_LIST authorities = proj_get_authorities_from_database( context );
+  PROJ_STRING_LIST authoritiesIt = authorities;
+  while ( char *authority = *authoritiesIt )
+  {
+    PROJ_STRING_LIST codes = proj_get_codes_from_database( context, authority, PJ_TYPE_ELLIPSOID, 0 );
+    PROJ_STRING_LIST codesIt = codes;
+    while ( char *code = *codesIt )
+    {
+      EllipsoidDefinition def;
+
+      PJ *ellipsoid = proj_create_from_database( context, authority, code, PJ_CATEGORY_ELLIPSOID, 0, nullptr );
+
+      QString name = QString( proj_get_name( ellipsoid ) );
+      def.acronym = QStringLiteral( "%1:%2" ).arg( authority, code );
+      name.replace( '_', ' ' );
+      def.description = QStringLiteral( "%1 (%2:%3)" ).arg( name, authority, code );
+
+      double semiMajor, semiMinor, invFlattening;
+      int semiMinorComputed = 0;
+      if ( proj_ellipsoid_get_parameters( context, ellipsoid, &semiMajor, &semiMinor, &semiMinorComputed, &invFlattening ) )
+      {
+        def.parameters.semiMajor = semiMajor;
+        def.parameters.semiMinor = semiMinor;
+        def.parameters.inverseFlattening = invFlattening;
+        if ( !semiMinorComputed )
+          def.parameters.crs = QgsCoordinateReferenceSystem::fromProj4( QStringLiteral( "+proj=longlat +a=%1 +b=%2 +no_defs" ).arg( def.parameters.semiMajor, 0, 'g', 17 ).arg( def.parameters.semiMinor, 0, 'g', 17 ) );
+        else if ( !qgsDoubleNear( def.parameters.inverseFlattening, 0.0 ) )
+          def.parameters.crs = QgsCoordinateReferenceSystem::fromProj4( QStringLiteral( "+proj=longlat +a=%1 +rf=%2 +no_defs" ).arg( def.parameters.semiMajor, 0, 'g', 17 ).arg( def.parameters.inverseFlattening, 0, 'g', 17 ) );
```

Comment:
I can't find a reference to `+rf` in the docs, where did you get that from?

Reply:
From the highest authority there is: http://osgeo-org.1560.x6.nabble.com/PROJ-Proj-6-API-questions-tp5398663p5398694.html 😆